compiler/p/codegen/OMRMemoryReference.cpp

```diff
@@ -354,8 +354,9 @@ void OMR::Power::MemoryReference::addToOffset(TR::Node * node, int64_t amount, T
       {
       self()->consolidateRegisters(NULL, NULL, false, cg);
       }
-   intptr_t displacement = self()->getOffset() + amount;
-   if (displacement<LOWER_IMMED || displacement>UPPER_IMMED)
+   int64_t displacement = self()->getOffset() + amount;
+   if (((displacement<LOWER_IMMED || displacement>UPPER_IMMED) && !cg->comp()->target().cpu.isAtLeast(OMR_PROCESSOR_PPC_P10)) ||
+         (displacement < -0x200000000 || displacement > 0x1ffffffff))
```

Comment:
Suggest:  making those hard-coded constants into macro defines as LOWER/UPPER_IMMED. They can be used in else where.  Plus, another flavor of them is expected soon (28-bit limit).

Reply:
Agree, definitely should make them macros.

Reply:
Fixed in 7dd94a5eddd89d1164aebe3ea55fe879357ae5d9